### ISSUE TITLE
Low zero-shot ImageNet acc with CC12M and ViT-B-32

### ISSUE THREAD

**Issue by ankitkv** [mlfoundations/open_clip#464]
For quick and easy experimentation with CLIP training, I'm looking for a setting that is large enough that lessons could transfer to larger-scale settings, while still being small enough that experiments can be run in reasonable time on 2-4 A100 GPUs.

Currently, I am training on [CC12M](https://github.com/google-research-datasets/conceptual-12m) using the following command (for 2 GPUs, batch size 2048, accumulation frequency 8 to get a global training batch size of 32768):
```
torchrun --nproc_per_node 2 -m training.main \
    --save-frequency 1 \
    --zeroshot-frequency 1 \
    --report-to wandb \
    --log-every-n-steps 2 \
    --train-data "[...]/cc12m/{00000..01242}.tar" \
    --logs "[...]/logs/" \
    --train-num-samples 10030127 \
    --dataset-type webdataset \
    --imagenet-val "[...]/imagenet/val/" \
    --warmup 2000 \
    --batch-size 2048 \
    --wd 0.2 \
    --epochs 32 \
    --workers 4 \
    --accum-freq 8 \
    --model ViT-B-32 \
    --name "ViT-B-32-Baseline" \
    --save-most-recent \
    --resume latest \
    --precision amp_bfloat16 \
    --seed 0 \
    --grad-checkpointing \
    --local-loss \
    --gather-with-grad
```

I get 18% zero-shot top-1 ImageNet val accuracy by epoch 16, after which the zero-shot accuracy starts going down slightly. I understand based on the first paragraph of README.md that one can get 22.2% on a ResNet-50x4 with just CC3M (which is a smaller dataset than CC12M), which leads me to believe I'm doing something wrong/suboptimal.

One odd observation is that I am easily able to reach a batch size of 2048, and can potentially go higher but when I attempt to do so, the dataloader goes OOM. However, all examples on README.md indicate much smaller batch sizes like 256 or 320. Is it expected that I should be able to reach batch size 2048?

In general, what can I do to improve my zero-shot ImageNet accuracy in the setting described above? Are there any published optimal hyperparameters for training on smaller scales?

**Comment by rwightman** [comment]
@ankitkv this has been covered in past issues and discussions, it's expected, ViT models need more data, use resnet or convnext if you want/need to use smaller datasets

**Comment by ankitkv** [comment]
@rwightman Thank you for your response. Can you please comment on the following part of the issue text?

> One odd observation is that I am easily able to reach a batch size of 2048, and can potentially go higher but when I attempt to do so, the dataloader goes OOM. However, all examples on README.md indicate much smaller batch sizes like 256 or 320. Is it expected that I should be able to reach batch size 2048?

If this is normal, is there anything you can suggest to reduce the memory requirements of the dataloader so that I can increase the batch size further until I fill the GPU? Thank you!

**Comment by mitchellnw** [comment]
Yes that’s normal since you’re using grad checkpoint and a small model. Those examples don’t have grad checkpoint. There’s an FSDP PR which may allow you to go to higher batchsize, but not in main yet.
